### ISSUE TITLE
Docker build fails for PHP8.2 docker file - Windows related

### ISSUE THREAD

**Issue by risharde** [tripal/tripal#1934]
### Tripal Version

- 

### Drupal Version

- 

### PostgreSQL Version

- 

### PHP Version

8.2

### General Description

The dockerfile specific to PHP 8.2 currently fails to build at step 18
This was on Windows but I don't think this is Windows specific

### Steps to reproduce

docker build -f tripaldocker/Dockerfile-php8.2  -t tripalproject/tripaldocker:latest . 

### Error messages

```shell
docker build -f tripaldocker/Dockerfile-php8.2  -t tripalproject/tripaldocker:latest . 
[+] Building 1.1s (22/31)                                                                      docker:desktop-linux
 => [internal] load build definition from Dockerfile-php8.2                                                    0.0s
 => => transferring dockerfile: 10.19kB                                                                        0.0s 
 => [internal] load metadata for docker.io/library/php:8.2-apache-bullseye                                     0.5s 
 => [internal] load .dockerignore                                                                              0.0s
 => => transferring context: 2B                                                                                0.0s 
 => [internal] load build context                                                                              0.2s 
 => => transferring context: 214.00kB                                                                          0.2s 
 => [ 1/27] FROM docker.io/library/php:8.2-apache-bullseye@sha256:151e272c24a7d3b791274fbad2cbabf57d2ba1f3ed5  0.0s 
 => CACHED [ 2/27] COPY . /app                                                                                 0.0s
 => CACHED [ 3/27] RUN chmod -R +x /app && apt-get update 1> ~/aptget.update.log   && apt-get install git unz  0.0s 
 => CACHED [ 4/27] RUN mkdir -p /usr/share/man/man1 && mkdir -p /usr/share/man/man7                            0.0s 
 => CACHED [ 5/27] RUN if [ "16" > "13" ] ; then   apt-get install -y curl apt-transport-https gpg --yes -qq   0.0s 
 => CACHED [ 6/27] RUN DEBIAN_FRONTEND=noninteractive apt-get update   && DEBIAN_FRONTEND=noninteractive apt-  0.0s 
 => CACHED [ 7/27] RUN    /etc/init.d/postgresql start &&    psql --command "CREATE USER docker WITH SUPERUSE  0.0s 
 => CACHED [ 8/27] RUN mv /app/tripaldocker/default_files/postgresql/pg_hba.conf /etc/postgresql/16/main/pg_h  0.0s 
 => CACHED [ 9/27] RUN echo "listen_addresses='*'" >> /etc/postgresql/16/main/postgresql.conf   && echo "max_  0.0s 
 => CACHED [10/27] RUN mv "/usr/local/etc/php/php.ini-development" "/usr/local/etc/php/php.ini"                0.0s 
 => CACHED [11/27] RUN pecl install xdebug-3.2.1     && docker-php-ext-enable xdebug     && cat /app/tripaldo  0.0s
 => CACHED [12/27] RUN set -eux;     if command -v a2enmod; then     a2enmod rewrite;   fi;     savedAptMark=  0.0s 
 => CACHED [13/27] RUN {     echo 'opcache.memory_consumption=128';     echo 'opcache.interned_strings_buffer  0.0s 
 => CACHED [14/27] RUN echo 'memory_limit = 1028M' >> /usr/local/etc/php/conf.d/docker-php-memlimit.ini   &&   0.0s 
 => CACHED [15/27] WORKDIR /var/www/html                                                                       0.0s 
 => CACHED [16/27] RUN echo "ServerName localhost" >> /etc/apache2/apache2.conf                                0.0s 
 => CACHED [17/27] WORKDIR /var/www                                                                            0.0s 
 => ERROR [18/27] RUN chmod a+x /app/tripaldocker/init_scripts/composer-init.sh   && /app/tripaldocker/init_s  0.2s
------
 > [18/27] RUN chmod a+x /app/tripaldocker/init_scripts/composer-init.sh   && /app/tripaldocker/init_scripts/composer-init.sh:
0.178 /bin/sh: 1: /app/tripaldocker/init_scripts/composer-init.sh: not found
------
Dockerfile-php8.2:157
--------------------
 156 |     WORKDIR /var/www
 157 | >>> RUN chmod a+x /app/tripaldocker/init_scripts/composer-init.sh \
 158 | >>>   && /app/tripaldocker/init_scripts/composer-init.sh
 159 |
--------------------
ERROR: failed to solve: process "/bin/sh -c chmod a+x /app/tripaldocker/init_scripts/composer-init.sh   && /app/tripaldocker/init_scripts/composer-init.sh" did not complete successfully: exit code: 127
```

**Comment by risharde** [comment]
Same build error occurs on the php 8.3 docker file, I might take a dig at fixing this if I can figure it out

**Comment by dsenalik** [comment]
Good luck! I can't help with windows dockers 🙈

**Comment by risharde** [comment]
I'm still confused after spending a couple hours trying to debug this
I see that the file exists when I do:
159 |     RUN chmod a+x /app/init_scripts/composer-init.sh
 160 |     RUN ls -la /app/init_scripts/composer-init.sh && sleep 10
 161 | >>> RUN /app/init_scripts/composer-init.sh
 162 |
 163 |     ## Use composer to install Drupal.
--------------------
ERROR: failed to solve: process "/bin/sh -c /app/init_scripts/composer-init.sh" did not complete successfully: exit code: 127

I'm surprised docker acts differently here

**Comment by laceysanderson** [comment]
Hmmm... I can confirm this doesn't happen on Mac (Intel) or Linux. We have [automated builds that show this works in the github environment](https://github.com/tripal/tripal/actions/runs/10044624463) so that makes it hard to debug as it's not reproducible outside of Windows.

This is so weird. Can you show us the full output from your modified version with the ls included and the execution of the composer script in it's own run? Also try adding the following parameters to your build command: `--progress=plain --no-cache` to make sure it's not caching any steps and so we get all the output rather then it collapsing.

```
docker build -f tripaldocker/Dockerfile-php8.2 -t tripalproject/tripaldocker:latest --progress=plain --no-cache .
```

**Comment by risharde** [comment]
Thanks Lace, I'll work on this further, I was able to get more steps in and
created a windows dockerfile but I still haven't had a successful build -
will use your command to get those logs and post back as soon as I can!

**Comment by risharde** [comment]
Okay so I got both the PHP8.2 and PHP 8.3 builds working for windows - I've added it to the V5b pub importer branch because I'm currently working on it

The issue was that Windows seems to format some of the script files in a way that linux isn't able to properly read so I had to add dos2unix to reformat those files when the docker is building.

I had to create dockerfile's in the root of the repo which seems messy but it's needed because of the way windows/docker build is processing the paths and since it doesn't like the symbolic link method.

I'm happy to try to keep these updated since I'm running Windows and I am aware that most of you all use *nix systems but I suppose there might be someone who would be trying tripal out on a windows machine at some point so hopefully this will help them out.
